extensions/amp-app-banner/1.0/component/component.js

```diff
@@ -0,0 +1,123 @@
+import * as Preact from '#preact';
+import {useEffect, useMemo, useRef} from '#preact';
+import {ContainWrapper} from '#preact/component';
+import {useLocalStorage} from '#preact/hooks/useLocalStorage';
+import {platformService} from '#preact/services/platform';
+
+import {userAssert} from '#utils/log';
+
+import {getAndroidAppInfo} from './android';
+import {useStyles} from './component.jss';
+import {getIOSAppInfo} from './ios';
+
+/**
+ * The raw App Banner component; no platform-specific logic
+ * @param {BentoAppBannerDef.RawProps} props
+ * @return {JSX.Element}
+ * @constructor
+ */
+export function AppBanner({
+  children,
+  dismissButtonAriaLabel = 'Dismiss',
+  onDismiss,
+  onInstall,
+  ...rest
+}) {
+  const styles = useStyles();
+
+  const bannerRef = useRef(null);
+  useEffect(() => {
```

Comment:
I wonder if we need to use `useLayoutEffect` to wait for the first paint to attach the click event handler. But if this works from your testing, that's great. I just won't have expected it to work.

Reply:
`useEffect` is better here, since it does not mutate the DOM.  It will run synchronously, before the user has any opportunity to click the banner. 
I had to google it to make sure, but [here's a great description](https://kentcdodds.com/blog/useeffect-vs-uselayouteffect):
> if your effect is mutating the DOM (via a DOM node ref) and the DOM mutation will change the appearance of the DOM node between the time that it is rendered and your effect mutates it, then you don't want to use useEffect. You'll want to use useLayoutEffect. Otherwise the user could see a flicker when your DOM mutations take effect. This is pretty much the only time you want to avoid useEffect and use useLayoutEffect instead.